mithril-common/src/chain_reader/pallas_chain_reader.rs

```diff
@@ -0,0 +1,349 @@
+                if socket_path.exists() {
+                    fs::remove_file(&socket_path).expect("Previous socket removal failed");
+                }
+
+                let known_point = get_fake_specific_point();
+                let unix_listener = UnixListener::bind(socket_path.as_path()).unwrap();
+                let mut server = NodeServer::accept(&unix_listener, 10).await.unwrap();
+
+                let chainsync_server = server.chainsync();
+
+                chainsync_server.recv_while_idle().await.unwrap();
+
+                chainsync_server
+                    .send_intersect_found(known_point.clone(), Tip(known_point.clone(), 1337))
+                    .await
+                    .unwrap();
+
+                chainsync_server.recv_while_idle().await.unwrap();
+
+                if has_agency == HasAgency::No {
+                    chainsync_server.send_await_reply().await.unwrap();
+                }
+
+                match action {
+                    ServerAction::RollBackward => {
+                        chainsync_server
+                            .send_roll_backward(known_point.clone(), Tip(known_point.clone(), 1337))
+                            .await
+                            .unwrap();
+                    }
+                    ServerAction::RollForward => {
+                        let block = BlockContent(get_fake_raw_block());
+                        chainsync_server
+                            .send_roll_forward(block, Tip(known_point.clone(), 1337))
+                            .await
+                            .unwrap();
+                    }
+                }
+            }
+        })
+    }
+
+    #[tokio::test]
+    async fn get_next_chain_block_rolls_backward() {
+        let socket_path =
+            create_temp_dir("get_next_chain_block_rolls_backward").join("node.socket");
+        let known_point = get_fake_specific_point();
+        let server = setup_server(
+            socket_path.clone(),
+            ServerAction::RollBackward,
+            HasAgency::Yes,
+        )
+        .await;
+        let client = tokio::spawn(async move {
+            let mut chain_reader =
+                PallasChainReader::new(socket_path.as_path(), CardanoNetwork::TestNet(10));
+
+            chain_reader
+                .set_chain_point(&ChainPoint::from(known_point.clone()))
+                .await
+                .unwrap();
+
+            chain_reader.get_next_chain_block().await.unwrap().unwrap()
+        });
+
+        let (_, client_res) = tokio::join!(server, client);
+        let chain_block = client_res.expect("Client failed to get next chain block");
+        match chain_block {
+            ChainBlockNextAction::RollBackward { rollback_point } => {
+                assert_eq!(rollback_point, get_fake_chain_point_backwards());
+            }
+            _ => panic!("Unexpected chain block action"),
+        }
+    }
+
+    #[tokio::test]
+    async fn get_next_chain_block_rolls_forward() {
```

Comment:
It's a bit difficult to understand what we try to test.
In one hand, we check the communication between client and server.
In another hand, we check the mapping from the server response to a ChainBlockNextAction.
Could we test this mapping (process_chain_block_next_action) in a another test to simply it ?

Reply:
I guess that we try to test that when the `chainsync` server returns a roll forward the `PallasChainReader::get_next_chain_block` function returns a `ChainBlockNextAction::RollForward` correctly formed :thinking: 

Could you elaborate on how we could split the test?